docs/src/configuration/app/source-operations.md

```diff
@@ -25,42 +40,160 @@ source:
+  <li>Click <strong>Run source operation</strong>.</li>
+  <li>Select the operation you want to run.
+  <li>(Optional) Add the <a href="#use-variables-in-your-source-operations">variables</a> required by your source operation.</li>
+  <li>Click <strong>Run</strong>.</li>
+</ol>
+
+<--->
+---
+title=Using the CLI
+file=none
+highlight=false
+---
+
+Run the following command:
 
 ```bash
-platform source-operation:run update-file
+platform source-operation:run <OPERATION_NAME>
 ```
 
-The `source-operation:run` command takes the command name (i.e. `update-file`) to run.
-Additional variables can be added to inject into the environment of the source operation.
-They will be interpreted the same way as any other [variable](../../development/variables/_index.md) set through the management console or the CLI,
-which means you need an `env:` prefix to expose them as a Unix environment variable.
+Replace `<OPERATION_NAME>` with the name of your operation, such as `update-file`) in the [example above](#1-define-a-source-operation).
+
+{{< /codetabs >}}
+
+## How source operations work
+
+When you trigger a source operation, the following happens in order:
+
+1. The current environment HEAD commit is checked out in Git.
+   It doesn't have any remotes or tags defined in the project.
+   It only has the current environment branch.
+2. Sequentially, for each app that has an operation with this name in its configuration,
+   the operation command is run in the app container.
+   The container isn't part of the environment's runtime cluster
+   and doesn't require that the environment is running.
+
+   The environment has all of the variables normally available during the build phase.
+   These may be optionally overridden by the variables specified when the operation is run.
+3. If any new commits were created, they're pushed to the repository and the normal build process is triggered.
+
+   If multiple apps in a single project both result in a new commit,
+   there are two distinct commits in the Git history but only a single new build process.
+
+## Use variables in your source operations
+
+You can add variables to the environment of the source operation.
+
+They're interpreted the same way as any other [variable](../../development/variables/_index.md)
+already set in your project.
+That means you need an `env:` prefix to expose them as a Unix environment variable.
+
 They can then be referenced by the source operation like any other variable.
 
+Say you wanted to have a `FILE` variable available with the value `example.txt`.
+Run the operation with the variable:
+
+{{< codetabs >}}
+
+---
+title=In the Console
+file=none
+highlight=false
+---
+
+<!--This is in HTML to get the icon not to break the list. -->
+<ol>
+  <li>Navigate to the environment where you want to run the operation.</li>
+  <li>Click {{< icon more >}} <strong>More</strong>.</li>
+  <li>Click <strong>Run source operation</strong>.</li>
```

Comment:
This might be changed. [More context in (3)](https://github.com/orgs/platformsh/projects/3/views/1)

Reply:
We can change in the future if it does change. For now, following [Google recommendations](https://developers.google.com/style/ui-elements?hl=en#formatting)